### PR TITLE
[MRG] Rpbi

### PR DESCRIPTION
**Pull request by bthirion**
This PR is meant to provide the RPBI methods in a Nilearn compatible way, without integrating it yet.
Seems to be working well on my box.

### CODE REVIEW

**Comment by bthirion** [comment]
Should I merge it ?

**Comment by AlexandreAbraham** [comment]
I you don't want reviews, you can do it!

**Comment by bthirion** [comment]
I'm fine with reviews!
I'm just afraid that this PR gets lost/forgotten as the nilearn PR did.
Note: the point is to close the Nilearn PR, which cannot stay in the 
limbs forever.

File: nilearn_sandbox/mass_univariate/rpbi.py [hunk 1]
```diff
@@ -0,0 +1,851 @@
...
+    # required. In scipy versions < 0.11, this import ends up
+    # importing matplotlib.pyplot. This sets the matplotlib backend
+    # which causes our matplotlib backend setting code in
+    # nilearn/plotting/__init__.py to have no effect. In environment
+    # without X, e.g. travis-ci, that means the tests will fail with
+    # the usual "TclError: no display name and no $DISPLAY environment
+    # variable". Note this is dependent on the order of import,
+    # whichever comes first has the only shot at setting the
+    # matplotlib backend.
+    from sklearn.cluster import FeatureAgglomeration
```
Comment (AlexandreAbraham) [inline nilearn_sandbox/mass_univariate/rpbi.py]: This delayed import may not be required anymore, if I understand the comment correctly.

File: nilearn_sandbox/mass_univariate/rpbi.py [hunk 2]
```diff
@@ -0,0 +1,851 @@
...
+    draw = rng.randint(n_samples, size=n_bootstrap_samples * n_parcellations)
+    draw = draw.reshape((n_parcellations, -1))
+    ret = joblib.Parallel(n_jobs=n_jobs)(joblib.delayed(
+            cache(_ward_fit_transform, memory, verbose=verbose))
+          (all_subjects_data, draw[i], connectivity, n_parcels, i * n_parcels)
+          for i in range(n_parcellations))
+    # reduce results
+    parcelled_data_parts, ward_labels = zip(*ret)
+    parcelled_data = np.hstack((parcelled_data_parts))
+    ward_labels = np.ravel(ward_labels)
```
Comment (AlexandreAbraham) [inline nilearn_sandbox/mass_univariate/rpbi.py]: ravel is executed twice, which is unnecessary I think.

**Comment by AlexandreAbraham** [comment]
I did not check everything and some optimizations are a bit opaque but, apart from that, LGTM.

**Comment by bthirion** [comment]
I think I addressed your comments. Thx !

**Pull request merged**